### PR TITLE
DOP-1620

### PR DESCRIPTION
**Pull request by casthewiz**
### Stories/Links:
[DOP-1620](https://jira.mongodb.org/browse/DOP-1620)

### Staging Links:
https://docs-mongodbcom-staging.corp.mongodb.com/master/datalake/cassidy.schaufele/DOP-1620/

### Notes:
- Replaces native html `abbr` with `inline-definition` component from LeafyGreen inside of `Abbr` component implementation
- ridealong adds a pull request template to the repo

#### New Implementation:
Screen Shot 2021-03-26 at 4 58 47 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,8 +38,13 @@
...
     "version": "auto-changelog -p --template keepachangelog && git add CHANGELOG.md"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
   "lint-staged": {
-    "*.js": [
+    "*": [
```
Comment (casthewiz) [inline package.json:47]: We currently run both 
```
"npm run format:fix",
"npm run lint:fix"
```
in `.github/workflows/test.yml` on every pull request - it makes sense to me to run the pre-commit hook for any file format changes then - this way other formats don't slip through the cracks (.md files, for example)

Totally open for discussion 😄

File: .eslintrc.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ module.exports = {
     browser: true,
   },
   extends: ['react-app', 'plugin:import/errors'],
-  ignorePatterns: ['docs-tools/', 'node_modules/', 'public/'],
+  ignorePatterns: ['docs-tools/', 'node_modules/', 'public/', '*.json'],
```
Comment (casthewiz) [inline .eslintrc.js:7]: With precommit hooks enabled, we were running eslint on `.json` files. We could take a closer look at the file format matching rules in the package.json, but this seemed like the quickest way to stop them from being false negatives.

**Comment by casthewiz** [comment]
The changes I've made in package.json allow husky to run linting as a pre-commit hook - essentially, we won't be able to commit unless the linting rules run and succeed, similar to our current linting in CI via github actions. 

**Failure State**
Screen Shot 2021-03-29 at 8 58 03 AM

**Success State**
Screen Shot 2021-03-29 at 8 43 32 AM

Very open to discussion & thoughts around this 😄

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+### Stories/Links:
+
+[Story Number](Url of Story)
```
Comment (sophstad) [inline .github/pull_request_template.md:3]: We use GitHub's [autolinked references](https://docs.github.com/en/github/writing-on-github/autolinked-references-and-urls) for our JIRA project, so only the ticket number is really necessary. Maybe something like this?
```suggestion
DOP-NNNN
```

**Review by sophstad — APPROVED** [review]
Sweet, thanks for adding the PR template! Also just fyi, titling the PR with the ticket name at the beginning (e.g. "DOP-1620 [Optional details]") will allow the JIRA webhook to pick it up.

**Review by rayangler — CHANGES_REQUESTED** [review]

File: src/components/Roles/Abbr.js [hunk 4]
```diff
@@ -15,7 +16,7 @@ const Abbr = ({
   if (expansion) {
     expansion = expansion.split(')')[0];
   }
-  return <abbr title={expansion}>{abbr}</abbr>;
+  return <InlineDefinition definition={expansion}>{abbr}</InlineDefinition>;
```
[the remark this answers is not in the document]
Comment (casthewiz) [reply src/components/Roles/Abbr.js:19]: Mmh, good catch

I'll add a test around that as well 👍

File: src/components/Roles/Abbr.js [hunk 5]
```diff
@@ -14,6 +14,7 @@ const Abbr = ({
   let [abbr, expansion] = value.split('(');
   if (expansion) {
     expansion = expansion.split(')')[0];
+    abbr = abbr.trim();
   }
   return <abbr title={expansion}>{abbr}</abbr>;
```
Comment (rayangler) [inline src/components/Roles/Abbr.js:19]: The last commit seems to have removed the LeafyGreen component!

**Review by rayangler — APPROVED** [review]
LGTM!

**Pull request merged**